src/CrossEncoderSearch.py

```diff
@@ -0,0 +1,19 @@
+from sentence_transformers import CrossEncoder
+
+class CrossencoderSearch:
+
+    def __init__(self,query,retrieved_docs):
+
+        self.query=query
+        self.pairs=retrieved_docs
+        self.final_pairs=[]
+        self.cross_encoding_model= CrossEncoder('BAAI/bge-base-en-v1.5')
+
+    def run_cross_encoder(self):
+        self.final_pairs = [[self.query, doc_text, self.cross_encoding_model.predict([self.query, doc_text])] for doc_text in self.pairs]
+        self.final_pairs.sort(key=lambda x: x[2])
+        self.final_pairs.reverse()
+        return self.final_pairs
+        
+def dict_to_list(lst):
+    return [''.join(d['Context Before']) + d['Main Sentence'] + ''.join(d['Context After']) for d in lst]
```

Comment:
before the context after ?

Reply:
'. '.join(d['Context Before'])
'. '.join(d['Context After'])

We want the sentences to be delimited by full stop or another delimiter. Otherwise they'll all become one big sentence

Reply:
shouldn't it only be on after since context before it the start of each string and it will have it's own full stops that differentiates it from the main sentences